runelite-api/src/main/java/net/runelite/api/Client.java

```diff
@@ -135,13 +135,29 @@
 	 */
 	GameState getGameState();
 
+	/**
+	 * Gets the current game state as an int
+	 *
+	 * @return the game state
+	 */
+	int getRSGameState();
+
 	/**
 	 * Sets the current game state
 	 *
 	 * @param gameState
 	 */
 	void setGameState(GameState gameState);
 
+	/**
+	 * Sets the current game state
+	 * This takes an int instead of a {@link GameState} so it can
+	 * can handle states that aren't in the enum yet
+	 *
+	 * @param gameState
+	 */
+	void setRSGameState(int gameState);
```

Comment:
make this into a mixin with a client thread check at the least, although this does just seem very dangerous overall

Reply:
Done. I wanted to add this is because we currently don't have a way to set the game state to states not in the GameState enum so it adds some functionality.

Reply:
Makes sense, but then again the gamestates could just be added to the enum where they could be properly documented.

Reply:
This adds a function to get the game state as an int so it can be added to the enum. There isn't currently a way to do this, so adding game states to the enum is very difficult.